react-components/src/components/RevealToolbar/LayersButton/components/LayersButtonsStrip.test.tsx

```diff
@@ -0,0 +1,147 @@
+const renderTargetMock = createRenderTargetMock();
+
+export const wrapper = ({ children }: { children: any }): any => (
+  <SDKProvider sdk={sdkMock}>
+    <QueryClientProvider client={queryClient}>
+      <ViewerContextProvider value={renderTargetMock}>
+        <Reveal3DResourcesInfoContextProvider>{children}</Reveal3DResourcesInfoContextProvider>
+      </ViewerContextProvider>
+    </QueryClientProvider>
+  </SDKProvider>
+);
+
+describe(LayersButtonStrip.name, () => {
+  const mockCadHandler = createCadHandlerMock();
+  const mockPointCloudHandler = createPointCloudHandlerMock();
+  const mockImage360Handler = createImage360HandlerMock();
+  const defaultProps: LayersButtonProps = {
+    layersState: {
+      cadLayers: [{ revisionId: 456, applied: true, index: 0 }],
+      pointCloudLayers: [{ revisionId: 123, applied: true, index: 0 }],
+      image360Layers: [{ siteId: 'site-id', applied: true }]
+    },
+    setLayersState: vi.fn(),
+    defaultLayerConfiguration: undefined
+  };
+
+  const defaultDependencies: LayersButtonDependencies = {
+    useModelHandlers: vi.fn((): [ModelLayerHandlers, () => void] => [
+      {
+        cadHandlers: [mockCadHandler],
+        pointCloudHandlers: [mockPointCloudHandler],
+        image360Handlers: [mockImage360Handler]
+      },
+      vi.fn()
+    ]),
+    useSyncExternalLayersState: vi.fn(),
+    ModelLayerSelection: vi.fn(({ label }) => <div>{label}</div>)
+  };
+
+  test('renders without crashing', () => {
+    const mockModels: CogniteModel[] = [cadMock, cadMock];
+    viewerModelsMock.mockReturnValue(mockModels);
+    const mockImage360Collection = createImage360ClassicMock();
+    viewerImage360CollectionsMock.mockReturnValue([mockImage360Collection]);
+
+    const { getAllByRole } = render(<LayersButtonStrip {...defaultProps} />, {
+      wrapper
+    });
+
+    expect(getAllByRole('img', { name: 'CubeIcon' })).toBeTruthy();
+    expect(getAllByRole('img', { name: 'PointCloudIcon' })).toBeTruthy();
+    expect(getAllByRole('img', { name: 'View360Icon' })).toBeTruthy();
+  });
+
+  test('should update viewer models visibility when layersState changes', () => {
+    const mockModels: CogniteModel[] = [cadMock, cadMock];
+    viewerModelsMock.mockReturnValue(mockModels);
+    const mockImage360Collection = createImage360ClassicMock();
+    viewerImage360CollectionsMock.mockReturnValue([mockImage360Collection]);
+    const ModelLayerSelection = vi.fn(({ label }) => <div>{label}</div>);
+    const newProps: LayersButtonDependencies & {
+      setLayersState: typeof defaultProps.setLayersState;
+    } = {
+      setLayersState: defaultProps.setLayersState,
+      ...defaultDependencies,
+      useModelHandlers: vi.fn(() => [
+        {
+          cadHandlers: [mockCadHandler],
+          pointCloudHandlers: [mockPointCloudHandler],
+          image360Handlers: [mockImage360Handler]
+        },
+        () => {}
+      ]) as unknown as LayersButtonDependencies['useModelHandlers'],
```

Comment:
is the `as unknown as LayersButton...` nessesary? You should be able to pass a generic to `vi.fn<...>`